lib/stac-auth-proxy/runtime/Dockerfile

```diff
@@ -0,0 +1,13 @@
+ARG PYTHON_VERSION
+FROM --platform=linux/amd64 public.ecr.aws/lambda/python:${PYTHON_VERSION}
+
+WORKDIR /tmp
+RUN python -m pip install pip -U
```

Comment:
If we used **uv** we'd get dependency locking for (almost) free ... is this something we want (I kind of think yes, tracking downstream issues might be easier if we lock our dependency tree)?

Reply:
I like this idea, but I think it would be worthy of a separate PR.